libcudacxx/test/libcudacxx/std/language.support/support.srcloc/general_nvrtc.pass.cpp

```diff
@@ -0,0 +1,116 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: !nvrtc
+
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+#include <cuda/std/source_location>
+#include <cuda/std/type_traits>
+
+#include "test_macros.h"
+
+static_assert(cuda::std::is_nothrow_move_constructible<cuda::std::source_location>::value, "support.srcloc.cons (1.1)");
+static_assert(cuda::std::is_nothrow_move_assignable<cuda::std::source_location>::value, "support.srcloc.cons (1.2)");
+#if TEST_STD_VER >= 2014
+static_assert(cuda::std::is_nothrow_swappable<cuda::std::source_location>::value, "support.srcloc.cons (1.3)");
+#endif // TEST_STD_VER >= 2014
+
+ASSERT_NOEXCEPT(cuda::std::source_location());
+#if !defined(TEST_COMPILER_NVCC)
+ASSERT_NOEXCEPT(cuda::std::source_location::current());
+#endif // TEST_COMPILER_NVCC
+
+__host__ __device__ bool compare_strings(const char* lhs, const char* rhs) noexcept
+{
+  for (size_t index = 0;; ++index)
+  {
+    if (lhs[index] != rhs[index])
+    {
+      return false;
+    }
+
+    if (lhs[index] == '\0')
+    {
+      return true;
+    }
+  }
+}
```

Comment:
Is this code generated? Otherwise we'd do good to factor it in somewhere.

Reply:
We have a overhaul of `char_traits` on our plate, so I am postponing that until then